File: frontend/src/types/poc.ts
export type POCStatus = 
  | 'Draft'
  | 'Planned'
  | 'In Progress'
  | 'On Hold'
  | 'Under Review'
  | 'Completed'
  | 'Successful'
  | 'Partially Successful'
  | 'Failed'
  | 'Archived'
  | 'Cancelled';

export type POCPriority = 'Low' | 'Medium' | 'High' | 'Critical';

export interface POC {
  id: string;
  poc_code: string;
  title: string;
  short_description: string;
  detailed_description: string;
  business_problem: string;
  technical_problem: string;
  objective: string;
  scope: string;
  out_of_scope?: string;
  category: string;
  priority: POCPriority;
  status: POCStatus;
  owner_id: string;
  team_id: string;
  technology_stack: string[];
  architecture_details?: string;
  evaluation_criteria?: any;
  test_results?: string;
  performance_results?: string;
  cost_estimation?: string;
  scalability_findings?: string;
  security_findings?: string;
  limitations?: string;
  risks?: string;
  recommendations?: string;
  outcome?: string;
  production_readiness?: string;
  lessons_learned?: string;
  ai_summary?: string;
  start_date?: string;
  expected_end_date?: string;
  completed_date?: string;
  is_archived: boolean;
  created_at: string;
  updated_at: string;
}

export interface POCDocument {
  id: string;
  poc_id: string;
  file_name: string;
  file_type: string;
  file_size: number;
  storage_url: string;
  version: number;
  description?: string;
  uploaded_by: string;
  processing_status: string;
  indexing_status: string;
  created_at: string;
}

File: backend/app/schemas/poc.py
from typing import Optional, List, Any
from datetime import date, datetime
from uuid import UUID
from pydantic import BaseModel

class POCCreate(BaseModel):
    poc_code: str
    title: str
    short_description: str
    detailed_description: str
    business_problem: str
    technical_problem: str
    objective: str
    scope: str
    out_of_scope: Optional[str] = None
    category: str
    priority: str = "Medium"
    status: str = "Draft"
    team_id: UUID
    technology_stack: List[str] = []
    architecture_details: Optional[str] = None
    evaluation_criteria: Optional[Any] = None
    test_results: Optional[str] = None
    performance_results: Optional[str] = None
    cost_estimation: Optional[str] = None
    scalability_findings: Optional[str] = None
    security_findings: Optional[str] = None
    limitations: Optional[str] = None
    risks: Optional[str] = None
    recommendations: Optional[str] = None
    outcome: Optional[str] = None
    production_readiness: Optional[str] = None
    lessons_learned: Optional[str] = None
    start_date: Optional[date] = None
    expected_end_date: Optional[date] = None

class POCUpdate(BaseModel):
    title: Optional[str] = None
    short_description: Optional[str] = None
    detailed_description: Optional[str] = None
    status: Optional[str] = None
    priority: Optional[str] = None
    technology_stack: Optional[List[str]] = None
    architecture_details: Optional[str] = None
    test_results: Optional[str] = None
    performance_results: Optional[str] = None
    limitations: Optional[str] = None
    risks: Optional[str] = None
    recommendations: Optional[str] = None
    outcome: Optional[str] = None
    production_readiness: Optional[str] = None
    lessons_learned: Optional[str] = None
    completed_date: Optional[date] = None

class POCResponse(POCCreate):
    id: UUID
    owner_id: UUID
    ai_summary: Optional[str] = None
    is_archived: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class POCListResponse(BaseModel):
    total: int
    items: List[POCResponse]
